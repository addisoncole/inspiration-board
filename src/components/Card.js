import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {

  deleteOnClick = () => {
    this.props.deleteCardCallback(this.props.id);
  }

  render() {
    const text = this.props.text;
    let moji = this.props.emoji;
    if (moji !== undefined && moji !== null) {
        moji = emoji.getUnicode(moji);
    }

    return (
      <div className="card">
        <span className="card__content">
          <div className="card__content-text">{text}</div>
          <div className="card__content-emoji">{moji}</div>
        </span>
        <button className="card__delete" onClick={this.deleteOnClick}> X </button>
      </div>
    )
  }
}

Card.propTypes = {
  id: PropTypes.number,
  text: PropTypes.string,
  emoji: PropTypes.string,
  deleteCardCallback: PropTypes.func,
};

export default Card;
