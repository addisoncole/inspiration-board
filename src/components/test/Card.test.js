import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  test('that it matches an existing snapshot', () => {
    // First Mount the Component in the testing DOM
    // Arrange
    const wrapper = shallow( <Card key={1} text={"test"} emoji={"heart_eyes"} />);

    // Assert that it looks like the last snapshot
    expect(wrapper).toMatchSnapshot();
  });
});
