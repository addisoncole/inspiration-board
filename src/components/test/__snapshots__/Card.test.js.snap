// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it matches an existing snapshot 1`] = `
<div
  className="card"
>
  <span
    className="card__content"
  >
    <div
      className="card__content-text"
    >
      test
    </div>
    <div
      className="card__content-emoji"
    >
      😍
    </div>
  </span>
  <button
    className="card__delete"
    onClick={[Function]}
  >
     X 
  </button>
</div>
`;
